server/mailet/rate-limiter-redis/src/main/scala/org/apache/james/rate/limiter/redis/RedisRateLimiter.scala

```diff
@@ -63,8 +62,8 @@ class RedisRateLimiterFactory @Inject()(redisConfiguration: RedisConfiguration)
       val resourceBuilder = ClientResources.builder()
         .threadFactoryProvider(poolName => NamedThreadFactory.withName(s"redis-driver-$poolName"))
       redisConfiguration.ioThreads.foreach(value => resourceBuilder.ioThreadPoolSize(value))
-      redisConfiguration.workerThreads.foreach(value =>resourceBuilder.computationThreadPoolSize(value))
-      new RedisClusterRateLimiterFactory(RedisClusterClient.create(resourceBuilder.build(),
+      redisConfiguration.workerThreads.foreach(value => resourceBuilder.computationThreadPoolSize(value))
+      new RedisClusterRateLimiterFactory(RedisUtils.createRedisClusterClient(resourceBuilder.build(),
```

Comment:
This is not the only place `RedisClusterClient` is being created e.g. `RedisHealthCheck`.

Reply:
rebase PR to avoid conflict 
https://github.com/apache/james-project/pull/2224